### PR TITLE
add option for xored output with fix key

### PR DESCRIPTION
**Pull request by jonasschnelli**
~~clients counterpart will follow soon.~~

Client part:https://github.com/bitcoin/bitcoin/pull/6021

### CODE REVIEW

File: dns.c [hunk 1]
```diff
@@ -324,13 +324,31 @@ ssize_t static dnshandle(dns_opt_t *opt, const unsigned char *inbuf, size_t insi
...
+          }
+          memcpy(addr[n].data.v4,IPXORed, 4);
+        }
+        ret = write_record_a(&outpos, outend - auth_size, "", offset, CLASS_IN, opt->datattl, &addr[n]);
+      }
+      else if (addr[n].v == 6) {
+        if (opt->useXor == 1)
+        {
+          unsigned char IPXORed[6];
+          for (int i=0; i<6; i++) {
```
Comment (sipa) [inline dns.c]: IPv6 records are 16 bytes (128 bits).

File: dns.c [hunk 2]
```diff
@@ -324,13 +324,31 @@ ssize_t static dnshandle(dns_opt_t *opt, const unsigned char *inbuf, size_t insi
...
+            IPXORed[i]=addr[n].data.v6[i]^xorKey[i];
+          }
+          memcpy(addr[n].data.v4,IPXORed, 6);
```
Comment (sipa) [inline dns.c]: .v6 ?

**Comment by sipa** [comment]
I'm afraid we'll need support to have both the old and new behaviour available for a while - probably by linking them to separate domain names?

File: dns.c (hunk 2 above)
Comment (jonasschnelli) [inline dns.c]: Oops. Right!

File: dns.c (hunk 1 above)
Comment (jonasschnelli) [inline dns.c]: Right!

**Comment by jonasschnelli** [comment]
Updated. If somebody queries with a "x." in front of your given hostname, it will xor "encode" the response.

**Comment by sipa** [comment]
No consensus, closing.

**Pull request closed without merging**